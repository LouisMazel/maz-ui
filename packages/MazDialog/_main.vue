<template>
  <transition
    name="dialog-fade"
    @after-enter="afterEnter"
    @after-leave="afterLeave"
  >
    <div
      v-if="value"
      class="maz-dialog maz-dialog--mask"
      :class="{
        'maz-dialog--success': success,
        'maz-dialog--danger': danger
      }"
    >
      <div class="maz-dialog__wrapper flex align-center">
        <div
          v-click-outside="closeDialog"
          :style="widthStyle"
          class="maz-dialog__container flex direction-column"
        >
          <div
            v-if="!hideHeader"
            class="maz-dialog__header flex space-between align-center p-3"
          >
            <p class="fw-400 fs-20 m-0 w-100">
              <slot name="title">
                Header
              </slot>
            </p>

            <!-- Close button -->
            <transition name="fade">
              <div
                v-if="hasClose"
                class="flex close-modal"
                @click="$emit('input', false)"
              >
                <i class="material-icons">
                  close
                </i>
              </div>
            </transition>
          </div>
          <div
            class="maz-dialog__body p-3"
          >
            <slot>Content</slot>
          </div>
          <div
            v-if="!hideFooter"
            class="maz-dialog__footer flex align-end justify-end p-3"
          >
            <slot name="footer">
              <MazBtn
                color="default"
                outline
                size="md"
                @click="$emit('input', false)"
              >
                Close
              </MazBtn>
              <MazBtn
                v-if="!noValidation"
                class="ml-3"
                color="primary"
                size="md"
                @click="$emit('validate')"
              >
                Confirm
              </MazBtn>
            </slot>
          </div>
        </div>
      </div>
    </div>
  </transition>
</template>

<script>
  import vClickOutside from 'v-click-outside'

  export default {
    name: 'MazDialog',
    directives: {
      clickOutside: vClickOutside.directive
    },
    props: {
      value: { type: Boolean, default: false },
      maxWidth: { type: String, default: '500px' },
      persistent: { type: Boolean, default: false },
      hasClose: { type: Boolean, default: true },
      hideHeader: { type: Boolean, default: false },
      hideFooter: { type: Boolean, default: false },
      noValidation: { type: Boolean, default: false },
      success: { type: Boolean, default: false },
      danger: { type: Boolean, default: false }
    },
    computed: {
      widthStyle () {
        return {
          maxWidth: this.maxWidth
        }
      }
    },
    methods: {
      closeDialog () {
        if (!this.persistent) {
          this.$emit('input', false)
        }
      },
      afterEnter () {
        this.$emit('opened')
      },
      afterLeave () {
        this.$emit('closed')
      }
    }
  }
</script>

<style lang="scss" scoped>
  .maz-dialog {
    &--mask {
      position: fixed;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;
      z-index: 1050;
      width: 100%;
      height: 100%;
      background-color: rgba(#303144, 0.7);
      transition: all 0.3s ease;
      overflow-y: auto;
      overflow-x: hidden;
    }

    &__wrapper {
      vertical-align: middle;
      min-height: 100%;
      width: 100%;
    }

    &__body {
      max-height: 100vh;
      color: $text-color;
    }

    &__container {
      margin: 30px auto;
      background-color: #FFF;
      box-shadow: 0 2px 8px rgba(0, 0, 0, 0.33);
      transition: all 0.3s ease;
      border-radius: $border-radius;

      @media only screen and (max-width: $breakpoint-tablet) {
        width: 100%;
      }
    }

    &__header {
      background-color: $bg-color-dark;
      border-top-left-radius: $border-radius;
      border-top-right-radius: $border-radius;
      border: none;
      color: $text-color-dark;

      h5 {
        color: white;
        margin-top: 0;
        font-size: 1.25rem !important;
      }

      .close-modal i {
        font-size: 18px;
        color: #FFF;
        cursor: pointer;

        &:hover {
          font-weight: bold;
        }
      }
    }

    &--success {
      .maz-dialog__header {
        background-color: $success-color;
      }
    }

    &--danger {
      .maz-dialog__header {
        background-color: $danger-color;
      }
    }
  }

  /** Modal animation **/
  .dialog-fade-enter-active .maz-dialog__container {
    animation: dialog-fade-in 0.4s;
  }

  .dialog-fade-leave-active .maz-dialog__container {
    animation: dialog-fade-out 0.4s;
  }

  @keyframes dialog-fade-in {
    0% {
      transform: translate3d(0, -30px, 0);
      opacity: 0;
    }

    100% {
      transform: translate3d(0, 0, 0);
      opacity: 1;
    }
  }

  @keyframes dialog-fade-out {
    0% {
      transform: translate3d(0, 0, 0);
      opacity: 1;
    }

    100% {
      transform: translate3d(0, -30px, 0);
      opacity: 0;
    }
  }
</style>
