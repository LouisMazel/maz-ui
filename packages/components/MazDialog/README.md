# MazDialog

## Props

<!-- @vuese:MazDialog:props:start -->

| Name            | Description                                                           | Type      | Required | Default      |
| --------------- | --------------------------------------------------------------------- | --------- | -------- | ------------ |
| value           | `true` if dialog is open / `false` if is close                        | `Boolean` | `true`   | -            |
| maxWidth        | is the `max-width` of the dialog (number in pixels)                   |           | `false`  | null         |
| width           | is the `width` of the dialog (number in pixels)                       |           | `false`  | null         |
| persistent      | if is `true`, is not possible to close he dialog with a click outside | `Boolean` | `false`  | false        |
| noHeader        | remove the header                                                     | `Boolean` | `false`  | false        |
| noFooter        | remove the footer                                                     | `Boolean` | `false`  | false        |
| noClose         | remove the close button                                               | `Boolean` | `false`  | false        |
| noConfirm       | remove the confirm button                                             | `Boolean` | `false`  | false        |
| success         | add "success" style to the dialog                                     | `Boolean` | `false`  | false        |
| danger          | add "danger" style to the dialog                                      | `Boolean` | `false`  | false        |
| dark            | add "dark" style to the dialog                                        | `Boolean` | `false`  | false        |
| excludedClasses | exclude elements classes (elements sometimes can close the dialog)    | `Array`   | `false`  | Array        |
| fullsize        | make dialog fullsize                                                  | `Boolean` | `false`  | false        |
| title           | title of the dialog                                                   | `String`  | `false`  | Header title |

<!-- @vuese:MazDialog:props:end -->

## Events

<!-- @vuese:MazDialog:events:start -->

| Event Name | Description                           | Parameters      |
| ---------- | ------------------------------------- | --------------- |
| input      | sent when dialog is close             | Boolean `false` |
| opened     | sent when after dialog is open        | event           |
| closed     | sent when after dialog is close       | event           |
| confirm    | sent when you click on confirm button | event           |

<!-- @vuese:MazDialog:events:end -->

## Slots

<!-- @vuese:MazDialog:slots:start -->

| Name    | Description                    | Default Slot Content                                    |
| ------- | ------------------------------ | ------------------------------------------------------- |
| title   | Replace the title element text | `<p class="maz-dialog__header__title">Title header</p>` |
| default | Replace the content            | `<p>Content</p>`                                        |
| footer  | Replace the footer bar         | Two `<MazBtn />`                                        |

<!-- @vuese:MazDialog:slots:end -->
