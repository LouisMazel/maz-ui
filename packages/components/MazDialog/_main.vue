<template>
  <transition
    name="maz-dialog-fade"
    @after-enter="afterEnter"
    @after-leave="afterLeave"
  >
    <div
      v-if="value"
      class="maz-dialog maz-dialog--mask"
      :class="{
        'maz-dialog--success': success,
        'maz-dialog--danger': danger,
        'maz-dialog--fullsize': fullsize,
        'maz-is-dark': dark
      }"
    >
      <div class="maz-dialog__wrapper maz-flex maz-align-center">
        <div
          v-click-outside="vcoConfig"
          :style="widthStyle"
          class="maz-dialog__container maz-dialog-animation maz-flex maz-direction-column maz-bg-color maz-border-radius"
          @keydown.esc="closeDialog"
        >
          <div
            v-if="!noHeader"
            class="maz-dialog__header maz-flex maz-space-between maz-align-center maz-p-3"
          >
            <!-- Replace the title element text -->
            <slot name="title">
              <!-- `<p class="maz-dialog__header__title">Title header</p>` -->
              <p class="maz-dialog__header__title">
                {{ title }}
              </p>
            </slot>

            <div
              v-if="!noClose"
              class="maz-flex close-modal"
              @click="$emit('input', false)"
            >
              <i class="material-icons">
                close
              </i>
            </div>
          </div>
          <div
            class="maz-dialog__body maz-p-3 maz-text-color"
          >
            <!-- Replace the content -->
            <slot>
              <!-- `<p>Content</p>` -->
              <p>Content</p>
            </slot>
          </div>
          <div
            v-if="!noFooter"
            class="maz-dialog__footer maz-flex maz-align-end maz-justify-end maz-p-3"
          >
            <!-- Replace the footer bar -->
            <slot name="footer">
              <!-- Two `<MazBtn />` -->
              <MazBtn
                color="default"
                outline
                size="md"
                @click="closeDialog"
              >
                Close
              </MazBtn>
              <MazBtn
                v-if="!noConfirm"
                class="maz-ml-3"
                size="md"
                :color="buttonConfirmColor"
                @click="onConfirm($event)"
              >
                Confirm
              </MazBtn>
            </slot>
          </div>
        </div>
      </div>
    </div>
  </transition>
</template>

<script>
import vClickOutside from 'v-click-outside'
import MazBtn from '../MazBtn'

export default {
  name: 'MazDialog',
  components: { MazBtn },
  directives: {
    clickOutside: vClickOutside.directive
  },
  props: {
    // `true` if dialog is open / `false` if is close
    value: { type: Boolean, required: true },
    // is the `max-width` of the dialog in pixels
    maxWidth: { type: Number, default: 500 },
    // if is `true`, is not possible to close he dialog with a click outside
    persistent: { type: Boolean, default: false },
    // remove the header
    noHeader: { type: Boolean, default: false },
    // remove the footer
    noFooter: { type: Boolean, default: false },
    // remove the close button
    noClose: { type: Boolean, default: false },
    // remove the confirm button
    noConfirm: { type: Boolean, default: false },
    // add "success" style to the dialog
    success: { type: Boolean, default: false },
    // add "danger" style to the dialog
    danger: { type: Boolean, default: false },
    // add "dark" style to the dialog
    dark: { type: Boolean, default: false },
    // exclude elements classes (elements sometimes can close the dialog)
    excludedClasses: { type: Array, default: Array },
    // make dialog fullsize
    fullsize: { type: Boolean, default: false },
    // title of the dialog
    title: { type: String, default: 'Header title' }
  },
  data () {
    return {
      vcoConfig: {
        handler: this.closeDialog,
        middleware: this.preventClickOutside,
        events: ['click'],
        isActive: true
      }
    }
  },
  computed: {
    widthStyle () {
      const { fullsize, maxWidth } = this
      return {
        maxWidth: fullsize ? null : `${maxWidth}px`
      }
    },
    buttonConfirmColor () {
      return this.danger
        ? 'danger'
        : this.success
          ? 'success'
          : 'primary'
    }
  },
  methods: {
    preventClickOutside () {
      return !this.excludedClasses.includes(event.target.className)
    },
    closeDialog () {
      if (!this.persistent) {
        // sent when dialog is close
        // @arg Boolean `false`
        this.$emit('input', false)
      }
    },
    afterEnter (e) {
      // sent when after dialog is open
      // @arg event
      this.$emit('opened', e)
    },
    afterLeave (e) {
      // sent when after dialog is close
      // @arg event
      this.$emit('closed', e)
    },
    onConfirm (e) {
      // sent when you click on confirm button
      // @arg event
      this.$emit('confirm', e)
    }
  }
}
</script>
