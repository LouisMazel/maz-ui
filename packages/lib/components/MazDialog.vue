<template>
  <MazBackdrop
    v-bind="backdropAttrs"
    transition-name="modal-anim"
    @close="$emit('close', $event)"
    @open="$emit('open', $event)"
    @update:model-value="$emit('update:model-value', $event)"
  >
    <template #default="{ close }">
      <div
        class="m-dialog"
        :style="[{ width, maxWidth, maxHeight }]"
        :class="{ '--scrollable': scrollable }"
        v-bind="wrapperAttrs"
      >
        <!--
          @slot Header slot
            @binding {Function} close close function
        -->
        <slot name="header" :close="close">
          <div class="m-dialog-header" :class="{ '--has-title': $slots['title'] || title }">
            <h2 v-if="$slots['title'] || title" class="maz-my-0 maz-text-xl maz-font-semibold">
              <!--
                @slot Title slot in the header
              -->
              <slot name="title">{{ title }}</slot>
            </h2>

            <MazBtn
              v-if="!noClose"
              size="sm"
              class="m-dialog-closebtn"
              color="transparent"
              @click="close"
            >
              <MazIcon :src="XIcon" class="maz-h-5 maz-w-5" />
            </MazBtn>
          </div>
        </slot>
        <div class="m-dialog-content">
          <!--
            @slot Default content
              @binding {Function} close close function
          -->
          <slot :close="close"></slot>
        </div>
        <div v-if="$slots['footer']" class="m-dialog-footer">
          <!--
            @slot Footer slot
              @binding {Function} close close function
          -->
          <slot name="footer" :close="close"></slot>
        </div>
      </div>
    </template>
  </MazBackdrop>
</template>

<script lang="ts" setup>
  import MazBtn from './MazBtn.vue'
  import MazBackdrop from './MazBackdrop.vue'
  import MazIcon from './MazIcon.vue'
  import XIcon from './../modules/icons/x.svg'
  import type { StyleValue } from 'vue'
  import { computed, useAttrs } from 'vue'
  import type { HTMLAttributes } from 'vue'

  defineProps({
    /** Modal's title in  header */
    title: { type: String, default: undefined },
    /** Remove the close button in header */
    noClose: { type: Boolean, default: false },
    /** Modal's width */
    width: { type: String, default: '500px' },
    /** Modal's max-width */
    maxWidth: { type: String, default: '95vw' },
    /** Modal's max-height */
    maxHeight: { type: String, default: '95vh' },
    /**  Modal's content becomes scrollable - warning: a overflow is applied */
    scrollable: { type: Boolean, default: false },
  })

  defineEmits([
    /** Model binding */
    'update:model-value',
    /** Emitted when dialog is closed (after animation) */
    'close',
    /** Emitted when dialog is opened (after animation) */
    'open',
  ])

  const attrs = useAttrs()

  const backdropAttrs = computed(() => ({
    ...attrs,
    class: undefined,
    style: undefined,
  }))
  const wrapperAttrs = computed<{
    class?: HTMLAttributes['class']
  }>(() => ({
    class: attrs.class,
    style: attrs.style as StyleValue,
  }))
</script>

<style lang="postcss" scoped>
  .m-dialog {
    @apply maz-flex maz-flex-col maz-rounded maz-bg-color maz-text-normal;

    &-header {
      @apply maz-flex maz-items-center maz-justify-end maz-pl-6 maz-pr-2 maz-pt-2;

      &.--has-title {
        @apply maz-justify-between;
      }
    }

    &-footer {
      @apply maz-flex maz-items-center maz-justify-end maz-px-6 maz-pb-4;
    }

    &-content {
      @apply maz-flex-1 maz-p-6;

      &-icon {
        flex: 0 0 auto;
      }
    }

    &.--scrollable {
      .m-dialog-content {
        @apply maz-overflow-auto;
      }
    }

    &-closebtn {
      @apply maz-p-2 !important;
    }
  }
</style>
