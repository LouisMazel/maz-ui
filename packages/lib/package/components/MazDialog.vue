<template>
  <MazBackDrop
    v-bind="$attrs"
    transition="modal-anim"
    @update:model-value="$emit('update:model-value', $event)"
  >
    <template #default="{ close }">
      <div class="m-dialog-layout">
        <div
          class="m-dialog-layout-header"
          :class="{ '--has-title': $slots['title'] || title }"
        >
          <h2
            v-if="$slots['title'] || title"
            class="maz-my-0 maz-text-xl maz-font-semibold"
          >
            <slot name="title">{{ title }}</slot>
          </h2>
          <MazBtn
            color="transparent"
            size="mini"
            class="m-dialog-layout-closebtn"
            @click="close"
          >
            <MazIcon :src="XIcon" class="maz-cursor-pointer" />
          </MazBtn>
        </div>
        <div class="m-dialog-layout-content">
          <slot />
        </div>
        <div v-if="$slots['footer']" class="m-dialog-layout-footer">
          <slot name="footer"></slot>
        </div>
      </div>
    </template>
  </MazBackDrop>
</template>

<script lang="ts" setup>
  import MazBtn from './MazBtn.vue'
  import MazBackDrop from './MazBackDrop.vue'
  import MazIcon from './MazIcon.vue'
  import XIcon from './../icons/x.svg'

  defineProps({
    title: { type: String, default: undefined },
  })

  defineEmits(['update:model-value'])

  // inheritAttrs: false,
</script>

<style lang="postcss" scoped>
  .m-dialog-layout {
    @apply maz-rounded-lg maz-bg-color maz-text-normal-text;

    width: 500px;
    max-width: 95vw;

    &-header {
      @apply maz-flex maz-items-center maz-justify-end maz-pt-2 maz-pl-6 maz-pr-2;

      &.--has-title {
        @apply maz-justify-between;
      }
    }

    &-footer {
      @apply maz-flex maz-items-center maz-justify-end maz-px-6 maz-pb-4;
    }

    &-content {
      @apply maz-p-6;

      &-icon {
        flex: 0 0 auto;
      }
    }

    &-closebtn {
      @apply maz-p-2 !important;
    }
  }
</style>
