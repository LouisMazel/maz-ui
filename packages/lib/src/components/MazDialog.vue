<script lang="ts" setup>
import type {
  ComponentPublicInstance,
  HTMLAttributes,
  StyleValue,
} from 'vue'
import type { MazBackdropProps } from './MazBackdrop.vue'

import { MazXMark } from '@maz-ui/icons'
import {
  computed,
  defineAsyncComponent,
  nextTick,
  ref,
  useAttrs,
  useSlots,
  watch,
} from 'vue'
import MazBackdrop from './MazBackdrop.vue'

defineOptions({
  name: 'MazDialog',
  inheritAttrs: false,
})

const props = withDefaults(defineProps<DialogProps & MazBackdropProps>(), {
  /** Title of the modal in header */
  title: undefined,
  /** Remove the close button in header */
  hideCloseButton: false,
  /** Modal's max-width */
  maxWidth: '100%',
  /** Modal's min-width */
  minWidth: '32rem',
  /**  Modal's content becomes scrollable - warning: a overflow is applied */
  scrollable: false,
  /** Persistent dialog (not closable by clicking outside and remove close button) */
  persistent: false,
})

defineEmits<{
  /** emitted when modal is open */
  'open': [value: void]
  /** emitted when modal is close */
  'close': [value: void]
  /** emitted when modal is open or close */
  'update:model-value': [value: boolean]
}>()

const MazBtn = defineAsyncComponent(() => import('./MazBtn.vue'))

export interface DialogProps {
  /** @model Modal's model value */
  modelValue?: boolean
  /** Title of the modal in header */
  title?: string
  /** Remove the close button in header */
  hideCloseButton?: boolean
  /** Modal's max-width */
  maxWidth?: string
  /** Modal's min-width */
  minWidth?: string
  /**  Modal's content becomes scrollable - warning: a overflow is applied */
  scrollable?: boolean
  /** Persistent dialog (not closable by clicking outside and remove close button) */
  persistent?: boolean
}

export interface MazDialogProps extends DialogProps, MazBackdropProps {}

const attrs = useAttrs()

const backdrop = ref<ComponentPublicInstance<typeof MazBackdrop>>()

defineExpose({
  close: () => backdrop.value?.close?.(),
})

const backdropProps = computed(() => {
  const dialogPropKeys: (keyof MazDialogProps)[] = [
    'modelValue',
    'title',
    'hideCloseButton',
    'maxWidth',
    'minWidth',
    'scrollable',
  ]

  return Object.fromEntries(
    Object.entries(props).filter(([key]) => !dialogPropKeys.includes(key as keyof MazDialogProps)),
  ) as MazBackdropProps
})

const backdropAttrs = computed(() => ({
  ...attrs,
  class: undefined,
  style: undefined,
}))
const wrapperAttrs = computed<{
  class?: HTMLAttributes['class']
}>(() => ({
  class: attrs.class,
  style: attrs.style as StyleValue,
}))

const dialogContent = ref<HTMLElement>()

const slots = useSlots()
const hasFooter = computed(() => !!slots.footer)

if (props.scrollable) {
  watch(() => props.modelValue, async (newVal) => {
    await nextTick()
    if (newVal && dialogContent.value) {
      setTimeout(() => {
        dialogContent.value?.scrollTo({ top: 0, behavior: 'instant' })
      }, 0)
    }
  })
}
</script>

<template>
  <MazBackdrop
    v-bind="{ ...backdropAttrs, ...backdropProps }"
    ref="backdrop"
    v-slot="{ close, onBackdropClicked }"
    :model-value="modelValue"
    transition-name="modal-anim"
    aria-labelledby="dialogTitle"
    aria-describedby="dialogDesc"
    content-padding
    justify="center"
    variant="dialog"
    @close="$emit('close', $event)"
    @open="$emit('open', $event)"
    @update:model-value="$emit('update:model-value', $event)"
  >
    <div
      class="m-dialog"
      role="dialog"
      aria-modal="true"
      :style="[{ '--max-width': maxWidth, '--min-width': minWidth }]"
      :class="{ '--scrollable': scrollable }"
      v-bind="wrapperAttrs"
    >
      <!--
        @slot Header slot
          @binding {Function} close close function
          @binding {Function} onBackdropClicked onBackdropClicked function (respects persistent option)
      -->
      <slot name="header" :close :on-backdrop-clicked>
        <div class="m-dialog-header" :class="{ '--has-title': $slots.title || title }">
          <h2
            v-if="$slots.title || title"
            id="dialogTitle"
            class="m-dialog-title"
          >
            <!--
                @slot Title slot in the header
              -->
            <slot name="title">
              {{ title }}
            </slot>
          </h2>

          <MazBtn
            v-if="!hideCloseButton && !persistent"
            color="transparent"
            size="sm"
            :icon="MazXMark"
            @click="close"
          />
        </div>
      </slot>
      <div id="dialogDesc" ref="dialogContent" class="m-dialog-content" :class="{ '--bottom-padding': !hasFooter }">
        <!--
            @slot Default content
              @binding {Function} close close function
              @binding {Function} onBackdropClicked onBackdropClicked function (respects persistent option)
          -->
        <slot :close :on-backdrop-clicked />
      </div>
      <div v-if="hasFooter" class="m-dialog-footer">
        <!--
            @slot Footer slot
              @binding {Function} close close function
              @binding {Function} onBackdropClicked onBackdropClicked function (respects persistent option)
          -->
        <slot name="footer" :close :on-backdrop-clicked />
      </div>
    </div>
  </MazBackdrop>
</template>

<style lang="postcss" scoped>
.m-dialog {
  @apply maz-flex maz-origin-center maz-flex-col maz-min-w-full maz-rounded maz-bg-color maz-text-normal dark:maz-border dark:maz-border-color-light tab-s:maz-my-8 maz-max-w-full maz-touch-none;

  @screen tab-s {
    max-width: var(--max-width);
    min-width: var(--min-width);
  }

  &-header {
    @apply maz-flex maz-items-baseline maz-justify-end maz-pl-6 maz-pr-2 maz-pt-2 maz-pb-4;

    &.--has-title {
      @apply maz-justify-between;
    }
  }

  &-title {
    @apply maz-my-0 maz-text-xl maz-font-semibold;
  }

  &-footer {
    @apply maz-flex maz-items-center maz-justify-end maz-px-6 maz-py-4;
  }

  &-content {
    @apply maz-flex-1 maz-px-6;

    &-icon {
      flex: 0 0 auto;
    }

    &.--bottom-padding {
      @apply maz-pb-4;
    }
  }

  &.--scrollable {
    @apply maz-max-h-[95vh] maz-my-0;

    .m-dialog-content {
      @apply maz-overflow-auto maz-border-t maz-border-color-light maz-py-4;

      &:not(.--bottom-padding) {
        @apply maz-border-b;
      }
    }
  }
}
</style>
