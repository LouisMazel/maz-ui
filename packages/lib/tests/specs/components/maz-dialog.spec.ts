import { type VueWrapper, shallowMount } from '@vue/test-utils'
import MazDialog from '@components/MazDialog.vue'

describe('mazDialog', () => {
  let wrapper: VueWrapper<InstanceType<typeof MazDialog>>

  beforeEach(() => {
    wrapper = shallowMount(MazDialog)
  })

  it('should match with the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
