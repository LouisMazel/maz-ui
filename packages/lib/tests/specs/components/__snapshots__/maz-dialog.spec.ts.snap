// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mazDialog > should match with the snapshot 1`] = `"<maz-backdrop-stub data-v-ac518804="" modelvalue="false" teleportselector="body" persistent="false" nocloseonesckey="false" transitionname="modal-anim" aria-labelledby="dialogTitle" aria-describedby="dialogDesc"></maz-backdrop-stub>"`;
